tmt/steps/provision/__init__.py

```diff
@@ -354,6 +354,9 @@ class TransferOptions:
     #: Ignore symlinks that point outside the source tree
     safe_links: bool = False
 
+    #: Run a ``mkdir -p`` of the destination before doing transfer
+    mkdir_p: bool = False
```

Comment:
Really dislike the name of this option. `create_destination`, or something at least remotely related to the semantics of this option?

Reply:
Fine with it, pending https://github.com/teemtee/tmt/pull/4064#discussion_r2358446993 resolution